src/manage_sql_nvts.c

```diff
@@ -347,17 +347,37 @@ insert_nvt (const nvti_t *nvti)
     g_warning ("%s: NVT with OID %s exists already, ignoring", __FUNCTION__,
                nvti_oid (nvti));
   else
-    sql ("INSERT into nvts (oid, name,"
-         " cve, bid, xref, tag, category, family, cvss_base,"
-         " creation_time, modification_time, uuid, solution_type,"
-         " qod, qod_type)"
-         " VALUES ('%s', '%s', '%s', '%s', '%s',"
-         " '%s', %i, '%s', '%s', %i, %i, '%s', '%s', %d, '%s');",
-         nvti_oid (nvti), quoted_name,
-         quoted_cve, quoted_bid, quoted_xref, quoted_tag,
-         nvti_category (nvti), quoted_family, quoted_cvss_base, creation_time,
-         modification_time, nvti_oid (nvti), quoted_solution_type,
-         qod, quoted_qod_type);
+    {
+      int i;
+
+      sql ("INSERT into nvts (oid, name,"
+           " cve, bid, xref, tag, category, family, cvss_base,"
+           " creation_time, modification_time, uuid, solution_type,"
+           " qod, qod_type)"
+           " VALUES ('%s', '%s', '%s', '%s', '%s',"
+           " '%s', %i, '%s', '%s', %i, %i, '%s', '%s', %d, '%s');",
+           nvti_oid (nvti), quoted_name,
+           quoted_cve, quoted_bid, quoted_xref, quoted_tag,
+           nvti_category (nvti), quoted_family, quoted_cvss_base, creation_time,
+           modification_time, nvti_oid (nvti), quoted_solution_type,
+           qod, quoted_qod_type);
+
+      sql ("DELETE FROM vt_refs where vt_oid = '%s';", nvti_oid (nvti));
```

Comment:
I think this should rather be done all at once in the caller, like with nvt_preferences (e.g. "TRUNCATE nvt_preferences;").  Or do you have something else planned?  These deletes are usually slow.

Reply:
I planned subsequent changes in separate PR's. Handling refs and prefs in a consistent way.

Reply:
OK, great.